src/compiler/scala/tools/nsc/javac/JavaParsers.scala

```diff
@@ -800,16 +800,8 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
       val superclazz =
         AppliedTypeTree(javaLangDot(tpnme.Enum), List(enumType))
       val finalFlag = if (enumIsFinal) Flags.FINAL else 0l
-      val abstractFlag = {
-        // javac adds `ACC_ABSTRACT` to enum classes with deferred members
-        val hasAbstractMember = body exists {
-          case d: DefDef => d.mods.isDeferred
-          case _         => false
-        }
-        if (hasAbstractMember) Flags.ABSTRACT else 0l
-      }
```

Comment:
The problem is that otherwise the pattern matcher–given  `public enum Foo { A, B }`–suggests that `Foo()` is missing in the exhaustiveness checks.

ClassfileParser does the same here: https://github.com/scala/scala/blob/2.12.x/src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala#L547